main/src/sphexa/CMakeLists.txt

```diff
@@ -8,9 +8,9 @@ function(enableH5Part exename)
 endfunction()
 
 set(exename sphexa)
-
 add_executable(${exename} sphexa.cpp)
 target_include_directories(${exename} PRIVATE ${SPH_EXA_INCLUDE_DIRS})
+target_include_directories(${exename} PRIVATE ${CMAKE_BINARY_DIR}/main/src)
```

Comment:
You could add `${CMAKE_BINARY_DIR}/main/src` to `SPH_EXA_INCLUDE_DIRS` a single time only (with PRIVATE)